src/Query.php

```diff
@@ -154,19 +154,92 @@ public function order($order, $overwrite = false)
         return $this;
     }
 
+    /**
+    * Sets the filter to use in a FilteredQuery object. Filters added using this method
+    * will be stacked on a bool filter and applied to the filter part of a filtered query.
+    *
+    * There are several way in which you can use this method. The easiest one is by passing
+    * a simple array of conditions:
+    *
+    * {{{
+    *   // Generates a {"term": {"name": "jose"}} json filter
+    *   $query->where(['name' => 'jose']);
+    * }}}
+    *
+    * You can have as many conditions in the array as you'd like, Operators are also allowe in
+    * the field side of the array:
+    *
+    * {{{
+    *   $query->where(['name' => 'jose', 'age >' => 30, 'interests in' => ['php', 'cake']);
+    * }}}
+    *
+    * You can read about the available operators and how they translate to Elastic Search
+    * filters in the `Cake\ElasticSearch\FilterBuilder::parse()` method documentation.
+    *
+    * Additionally, it is possible to use a closure as first argument. The closure will receive
+    * a FilterBuilder instance, that you can use for creating arbitrary filter combinations:
+    *
+    * {{{
+    *   $query->where(function ($builder) {
+    *    return $builder->and($builder->between('age', 10, 20), $builder->missing('name'));
+    *   });
+    * }}}
+    *
+    * Finally, you can pass any already built filters as first argument:
+    *
+    * {{{
+    *   $query->where(new Elastica\Filter\Term('name.first', 'jose'));
+    * }}{
+    *
+    * @param array|callable|Elastica\Filter\AbstractFilter $conditions The list of conditions.
+    * @param bool $overwrite Whether or not to replace previous filters.
+    * @return $this
+    * @see Cake\ElasticSearch\FilterBuilder
+    */
     public function where($conditions, $overwrite = false)
     {
-        if ($this->_parts['preFilter'] === null || $overwrite) {
-            $this->_parts['preFilter'] = new BoolFilter;
+        return $this->_buildFilter('preFilter', $conditions, $overwrite);
```

Comment:
preFilter is something I made up, it is not going to elastica or to elastic search